src/BeadingStrategy/CenterDeviationBeadingStrategy.cpp

```diff
@@ -11,58 +11,79 @@ CenterDeviationBeadingStrategy::Beading CenterDeviationBeadingStrategy::compute(
-            ret.toolpath_locations.emplace_back(optimal_width * (bead_idx * 2 + 1) / 2);
+            ret.bead_widths.emplace_back(optimal_width_inner);
+            ret.toolpath_locations.emplace_back(optimal_width_outer + optimal_width_inner * ((bead_idx - 1) * 2 + 1) / 2);
         }
+
+        // middle (if any):
+        const coord_t optimal_width_middle = bead_count == 1 ? optimal_width_outer : optimal_width_inner;
         if (bead_count % 2 == 1)
         {
-            ret.bead_widths.emplace_back(thickness - (bead_count - 1) * optimal_width);
+            ret.bead_widths.emplace_back(thickness - (optimal_width_outer + (bead_count - 2) * optimal_width_inner));
             ret.toolpath_locations.emplace_back(thickness / 2);
             ret.left_over = 0;
         }
         else
         {
-            ret.left_over = thickness - bead_count * optimal_width;
+            ret.left_over = thickness - bead_count * optimal_width_middle;
         }
+
+        // last half minus middle:
         for (coord_t bead_idx = (bead_count + 1) / 2; bead_idx < bead_count; bead_idx++)
         {
-            ret.bead_widths.emplace_back(optimal_width);
-            ret.toolpath_locations.emplace_back(thickness - (bead_count - bead_idx) * optimal_width + optimal_width / 2);
+            ret.bead_widths.emplace_back(optimal_width_inner);
+            ret.toolpath_locations.emplace_back(thickness - (optimal_width_outer + ((bead_count - bead_idx) - 1) * optimal_width_inner) + optimal_width_inner / 2);
         }
     }
     else
     {
         ret.left_over = thickness;
     }
+
     return ret;
 }
 
 coord_t CenterDeviationBeadingStrategy::getOptimalThickness(coord_t bead_count) const
 {
-    return bead_count * optimal_width;
+    return std::max(0LL, (bead_count - 2)) * optimal_width_inner + std::min(2LL, bead_count) * optimal_width_outer;
 }
 
 coord_t CenterDeviationBeadingStrategy::getTransitionThickness(coord_t lower_bead_count) const
 {
     if (lower_bead_count % 2 == 0)
     { // when we add the extra bead in the middle
-        return lower_bead_count * optimal_width + underfill_bound;
+        return this->getOptimalThickness(lower_bead_count) + underfill_bound;
     }
     else
     { // when we move away from the strategy which replaces two beads by a single one in the middle
-        return (lower_bead_count + 1) * optimal_width - overfill_bound;
+        return this->getOptimalThickness(lower_bead_count + 1) - overfill_bound;
     }
 }
 
 coord_t CenterDeviationBeadingStrategy::getOptimalBeadCount(coord_t thickness) const
 {
-    coord_t naive_count = (thickness / 2 + optimal_width / 2) / optimal_width * 2;
-    coord_t optimal_thickness = naive_count * optimal_width;
-    coord_t overfill = optimal_thickness - thickness;
+    coord_t thickness_left = thickness;
+    coord_t naive_count = 0;
+    naive_count += std::min(2LL, ((thickness / 2 + optimal_width_outer / 2) / optimal_width_outer) * 2);
```

Comment:
Why not initialize the naive_count immediately with this value?

Reply:
There was an if statement around it before I thought to use the std::min I think. I'm fixing these two right now, let me know if you find more.